### PR TITLE
Create documentation about upgrade

### PR DESCRIPTION
**Pull request by polskikiel**
**Description**

Changes proposed in this pull request:

- Created documentation about upgrade

**Related issue(s)**

### CODE REVIEW

File: docs/kyma-environment-broker/08-05-orchestrate-kyma-upgrade.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+title: Upgrade Kyma Runtime using KEB
+type: Tutorials
+---
+
+This tutorial shows how to upgrade Kyma Runtime using Kyma Environment Broker.
+
+## Prerequisites
+
+- Compass with:
+  * Runtime Provisioner [configured](/control-plane/runtime-provisioner/#tutorials-provision-clusters-through-gardener) for Azure provider
```
Comment (klaudiagrz) [inline docs/kyma-environment-broker/08-05-orchestrate-kyma-upgrade.md:11]:
```suggestion
  * Runtime Provisioner [configured](/control-plane/runtime-provisioner/#tutorials-provision-clusters-through-gardener) for Azure
```

File: docs/kyma-environment-broker/08-05-orchestrate-kyma-upgrade.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+A successful call returns the orchestration ID:
+
+   ```json
+   {
+       "orchestration_id":"8a7bfd9b-f2f5-43d1-bb67-177d2434053c"
+   }
+   ```
+
+4. Check the operation status as described [here](#tutorials-check-orchestration-status).
```
Comment (klaudiagrz) [inline docs/kyma-environment-broker/08-05-orchestrate-kyma-upgrade.md:54]:
```suggestion
4. [Check the operation status](#tutorials-check-orchestration-status).
```

File: docs/kyma-environment-broker/08-05-orchestrate-kyma-upgrade.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+## Steps
+
+1. Export these values as environment variables:
+
+   ```bash
+   export RUNTIME_ID={RUNTIME_ID}
+   ```
+
+2. Get the [access token](#details-authorization). Export this variable based on the token you got from the OAuth client:
```
Comment (klaudiagrz) [inline docs/kyma-environment-broker/08-05-orchestrate-kyma-upgrade.md:22]:
```suggestion
2. [Get the access token](#details-authorization). Export this variable based on the token you got from the OAuth client:
```

File: docs/kyma-environment-broker/08-05-orchestrate-kyma-upgrade.md [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+   ```bash
+   export AUTHORIZATION_HEADER="Authorization: Bearer $ACCESS_TOKEN"
+   ```
+
+3. Make a call to the Kyma Environment Broker to orchestrate the upgrade.
+
+>**NOTE:** **dry** parameter specified to in the request body to `true` causes that upgrade is executed in a fake mode. It means that actions are not executed against selected runtimes but the Orchestration status is still available.
```
Comment (klaudiagrz) [inline docs/kyma-environment-broker/08-05-orchestrate-kyma-upgrade.md:30]:
```suggestion
>**NOTE:** The **dry** parameter specified in the request body is set to `true`. This causes that the upgrade is executed in a fake mode. It means that all actions are not executed against selected Runtimes but the Orchestration status is still available.
```

Comment (klaudiagrz) [reply docs/kyma-environment-broker/08-05-orchestrate-kyma-upgrade.md:30]: I'd like some explanation about this fake mode 😄

**Review by klaudiagrz — COMMENTED** [review]
See the rough review and let's discuss the details.

File: docs/kyma-environment-broker/03-10-orchestration.md [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+
+Orchestration is an abstraction which allows to aggregate Kyma and cluster upgrades. User needs to send a request on a [proper KEB's handler](##tutorials-upgrade-kyma-runtime-using-keb) to create an Orchestration.
+
+After creation, orchestration is being processed by KymaUpgradeManager which resolves Shoots on the gardener cluster and filters them using input provided by the User in the request body.
+
+When Orchestration knows its targets, it starts to process them in a separate queue as the Upgrade Operation using the steps logic. 
+
+User can always check the status of the Orchestration and its Operations using one of the `GET` endpoints.
+
+>**NOTE:** For now only Kyma upgrade is supported
```
[the remark this answers is not in the document]
Comment (polskikiel) [reply docs/kyma-environment-broker/03-10-orchestration.md:14]: we'll support also cluster upgrades in the future

File: docs/kyma-environment-broker/03-10-orchestration.md [hunk 6]
```diff
@@ -0,0 +1,18 @@
+---
+title: Orchestration
+type: Details
+---
+
+Orchestration is an abstraction which allows to aggregate Kyma and cluster upgrades. User needs to send a request on a [proper KEB's handler](##tutorials-upgrade-kyma-runtime-using-keb) to create an Orchestration.
+
+After creation, orchestration is being processed by KymaUpgradeManager which resolves Shoots on the gardener cluster and filters them using input provided by the User in the request body.
+
+When Orchestration knows its targets, it starts to process them in a separate queue as the Upgrade Operation using the steps logic.
```
[the remark this answers is not in the document]
Comment (polskikiel) [reply docs/kyma-environment-broker/03-10-orchestration.md]: I paraphrased that sentence

File: docs/kyma-environment-broker/03-10-orchestration.md [hunk 7]
```diff
@@ -0,0 +1,18 @@
+---
+title: Orchestration
+type: Details
+---
+
+Orchestration is an abstraction that allows you to aggregate Kyma and cluster upgrades. To create an orchestration, [send a request to the proper KEB handler](#tutorials-upgrade-kyma-runtime-using-keb).
```
Comment (klaudiagrz) [inline docs/kyma-environment-broker/03-10-orchestration.md:6]:
```suggestion
Orchestration is a mechanism that allows you to upgrade Kyma Runtimes. 

To create an orchestration, [send a request to the proper KEB handler](#tutorials-upgrade-kyma-runtime-using-keb).
```

File: docs/kyma-environment-broker/03-10-orchestration.md [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+
+After sending the request, the orchestration is processed by KymaUpgradeManager which resolves Shoots in the Gardener cluster and filters them using input provided in the request body.
+
+When Orchestration knows its targets, it starts to process them in a separate queue as the Upgrade Operation leveraging the [upgrade steps](#details-runtime-operations) logic. 
+
+You can check the status of the Orchestration and its operations using one of the `GET` endpoints.
+
+>**NOTE:** For now only Kyma upgrade is supported
```
Comment (klaudiagrz) [inline docs/kyma-environment-broker/03-10-orchestration.md:14]:
```suggestion

```

Comment (klaudiagrz) [reply docs/kyma-environment-broker/03-10-orchestration.md:14]: I'd avoid promising something that is not yet implemented. Let's add info on other upgrade options when they will actually be available.

File: docs/kyma-environment-broker/03-10-orchestration.md [hunk 9]
```diff
@@ -0,0 +1,18 @@
+---
+title: Orchestration
+type: Details
+---
+
+Orchestration is an abstraction that allows you to aggregate Kyma and cluster upgrades. To create an orchestration, [send a request to the proper KEB handler](#tutorials-upgrade-kyma-runtime-using-keb).
+
+After sending the request, the orchestration is processed by KymaUpgradeManager which resolves Shoots in the Gardener cluster and filters them using input provided in the request body.
```
Comment (klaudiagrz) [inline docs/kyma-environment-broker/03-10-orchestration.md:8]:
```suggestion
After sending the request, the orchestration is processed by `KymaUpgradeManager` which lists Shoots (Kyma Runtimes) in the Gardener cluster and filters the Runtime IDs that you have specified for the upgrade in the request body.
```

File: docs/kyma-environment-broker/03-10-orchestration.md [hunk 10]
```diff
@@ -0,0 +1,18 @@
...
+
+When Orchestration knows its targets, it starts to process them in a separate queue as the Upgrade Operation leveraging the [upgrade steps](#details-runtime-operations) logic.
```
Comment (klaudiagrz) [inline docs/kyma-environment-broker/03-10-orchestration.md:10]:
```suggestion
Then, the `KymaUpgradeManager` proceeds with the orchestration and performs the [upgrade steps](#details-runtime-operations) logic on the Runtimes that you have specified. 
```

Comment (klaudiagrz) [reply docs/kyma-environment-broker/03-10-orchestration.md:10]: We can also mention here that for now, we can upgrade only one Runtime per request.

File: docs/kyma-environment-broker/03-10-orchestration.md [hunk 11]
```diff
@@ -0,0 +1,18 @@
...
+type: Details
+---
+
+Orchestration is an abstraction that allows you to aggregate Kyma and cluster upgrades. To create an orchestration, [send a request to the proper KEB handler](#tutorials-upgrade-kyma-runtime-using-keb).
+
+After sending the request, the orchestration is processed by KymaUpgradeManager which resolves Shoots in the Gardener cluster and filters them using input provided in the request body.
+
+When Orchestration knows its targets, it starts to process them in a separate queue as the Upgrade Operation leveraging the [upgrade steps](#details-runtime-operations) logic. 
+
+You can check the status of the Orchestration and its operations using one of the `GET` endpoints.
```
Comment (klaudiagrz) [inline docs/kyma-environment-broker/03-10-orchestration.md:12]:
```suggestion
To check the status of the orchestration, use one of the `GET` endpoints.
```

File: docs/kyma-environment-broker/03-10-orchestration.md [hunk 12]
```diff
@@ -0,0 +1,18 @@
...
+
+>**NOTE:** For now only Kyma upgrade is supported
+
+To create an Orchestrations, use a token with the `broker-upgrade:write` authorization scope. To fetch Orchestrations, use the `broker-upgrade:read` scope.
+
+Use the Swagger to [check how to consume the Kyma Orchestration API](https://app.swaggerhub.com/apis/kempski/kyma-orchestration_api/0.1).
```
Comment (klaudiagrz) [inline docs/kyma-environment-broker/03-10-orchestration.md:18]: Move this sentence under this sentence:

`To check the status of the orchestration, use one of the `GET` endpoints.`

File: docs/kyma-environment-broker/03-10-orchestration.md [hunk 13]
```diff
@@ -0,0 +1,18 @@
...
+
+After sending the request, the orchestration is processed by KymaUpgradeManager which resolves Shoots in the Gardener cluster and filters them using input provided in the request body.
+
+When Orchestration knows its targets, it starts to process them in a separate queue as the Upgrade Operation leveraging the [upgrade steps](#details-runtime-operations) logic. 
+
+You can check the status of the Orchestration and its operations using one of the `GET` endpoints.
+
+>**NOTE:** For now only Kyma upgrade is supported
+
+To create an Orchestrations, use a token with the `broker-upgrade:write` authorization scope. To fetch Orchestrations, use the `broker-upgrade:read` scope.
```
Comment (klaudiagrz) [inline docs/kyma-environment-broker/03-10-orchestration.md:16]:
```suggestion
>**NOTE:** You need a token with the `broker-upgrade:write` authorization scope to create an orchestration, and a token with the `broker-upgrade:read` scope to fetch the orchestrations.
```

File: docs/kyma-environment-broker/03-10-orchestration.md (hunk 9 above)
Comment (klaudiagrz) [reply docs/kyma-environment-broker/03-10-orchestration.md:8]: Please combine this section with the previous sentence.

File: docs/kyma-environment-broker/03-10-orchestration.md (hunk 13 above)
Comment (klaudiagrz) [reply docs/kyma-environment-broker/03-10-orchestration.md:16]: I'd move this sentence as NOTE under this sentence:

```After sending the request, the orchestration is processed by `KymaUpgradeManager` which lists Shoots (Kyma Runtimes) in the Gardener cluster and filters the Runtime IDs that you have specified for the upgrade in the request body.```

File: docs/kyma-environment-broker/03-03-runtime-operations.md [hunk 14]
```diff
@@ -50,9 +50,21 @@ The deprovisioning process contains the following steps:
...
+The upgrade process contains the following steps:
+
+| Name                         | Domain         | Status      | Description                                                                            | Owner     |
+|------------------------------|----------------|-------------|----------------------------------------------------------------------------------------|-----------|
+| Upgrade_Kyma_Initialisation  | Upgrade | Done        | Initializes the `UpgradeOperation` instance with data fetched from the `ProvisioningOperation`. | @ksputo (Team Gopher) |
+| Overrides_From_Secrets_And_Config_Step  | Upgrade | Done        | Builds an input configuration for the Provisioner. | @ksputo (Team Gopher) |
+| Upgrade_Runtime              | Upgrade | Done        | Triggers upgrading of a Runtime in the Runtime Provisioner. | @ksputo (Team Gopher) |
 
-You can configure Runtime provisioning and deprovisioning processes by providing additional steps. To add a new provisioning or deprovisioning step, follow these tutorials:
+## Add/provide additional steps
```
Comment (klaudiagrz) [inline docs/kyma-environment-broker/03-03-runtime-operations.md:65]: Please choose only one verb 😄

Comment (klaudiagrz) [reply docs/kyma-environment-broker/03-03-runtime-operations.md:65]: Also, there's something wrong with this doc. There are two Deprovisioning sections now. And still, the metadata section is not updated.

File: docs/kyma-environment-broker/08-05-orchestrate-kyma-upgrade.md [hunk 15]
```diff
@@ -0,0 +1,54 @@
...
+
+2. [Get the access token](#details-authorization). Export this variable based on the token you got from the OAuth client:
+
+   ```bash
+   export AUTHORIZATION_HEADER="Authorization: Bearer $ACCESS_TOKEN"
+   ```
+
+3. Make a call to the Kyma Environment Broker to orchestrate the upgrade.
+
+>**NOTE:** The **dry** parameter specified in the request body is set to `true`. This causes that the upgrade is executed in a fake mode. It means that all actions are not executed against selected Runtimes but the Orchestration status is still available.
```
Comment (klaudiagrz) [inline docs/kyma-environment-broker/08-05-orchestrate-kyma-upgrade.md:30]:
```suggestion
>**NOTE:** The **dry** parameter specified in the request body is set to `true`. As a result, the upgrade is executed in a fake mode. It means that all actions are not executed against selected Runtimes but the Orchestration status is still available.
```

File: docs/kyma-environment-broker/03-10-orchestration.md [hunk 16]
```diff
@@ -0,0 +1,13 @@
...
+---
+
+Orchestration is a mechanism that allows you to upgrade Kyma Runtimes. To create an orchestration, [send the upgrade request to the proper KEB handler](#tutorials-upgrade-kyma-runtime-using-keb). After sending the request, the orchestration is processed by `KymaUpgradeManager`. It lists Shoots (Kyma Runtimes) in the Gardener cluster and filters the Runtime IDs that you have specified for the upgrade in the request body. Then, the `KymaUpgradeManager` proceeds with the orchestration and performs the [upgrade steps](#details-runtime-operations) logic on the Runtimes that you have specified.
+
+>**NOTE:** You need a token with the `broker-upgrade:write` authorization scope to create an orchestration, and a token with the `broker-upgrade:read` scope to fetch the orchestrations.
+
+>**NOTE:** For now, you can upgrade only one Runtime in a single request in KEB.
+
+To check the status of the orchestration, use one of the `GET` endpoints.
+Use Swagger to [check how to consume the Kyma Orchestration API](https://app.swaggerhub.com/apis/kempski/kyma-orchestration_api/0.1).
```
Comment (klaudiagrz) [inline docs/kyma-environment-broker/03-10-orchestration.md:13]: I'd consider describing the API in more detail in this doc.

Comment (polskikiel) [reply docs/kyma-environment-broker/03-10-orchestration.md:13]: imo we can - list our API's endpoints and briefly describe them, and then we can link to the Swagger for more details.

Comment (klaudiagrz) [reply docs/kyma-environment-broker/03-10-orchestration.md:13]: I like this idea 🚀

File: docs/kyma-environment-broker/03-10-orchestration.md [hunk 17]
```diff
@@ -0,0 +1,18 @@
+---
+title: Orchestration
+type: Details
+---
+
+Orchestration is a mechanism that allows you to upgrade Kyma Runtimes. To create an orchestration, [send the upgrade request to the proper KEB handler](#tutorials-upgrade-kyma-runtime-using-keb). After sending the request, the orchestration is processed by `KymaUpgradeManager`. It lists Shoots (Kyma Runtimes) in the Gardener cluster and filters the Runtime IDs that you have specified for the upgrade in the request body. Then, the `KymaUpgradeManager` proceeds with the orchestration and performs the [upgrade steps](#details-runtime-operations) logic on the Runtimes that you have specified.
```
Comment (klaudiagrz) [inline docs/kyma-environment-broker/03-10-orchestration.md:6]:
```suggestion
Orchestration is a mechanism that allows you to upgrade Kyma Runtimes. To create an orchestration, [send the upgrade request to the proper KEB handler](#tutorials-upgrade-kyma-runtime-using-keb). After sending the request, the orchestration is processed by `KymaUpgradeManager`. It lists Shoots (Kyma Runtimes) in the Gardener cluster and filters the Runtime IDs that you have specified in the request body. Then, `KymaUpgradeManager` performs the [upgrade steps](#details-runtime-operations) logic on the Runtimes that you have specified.
```

File: docs/kyma-environment-broker/03-10-orchestration.md [hunk 18]
```diff
@@ -0,0 +1,18 @@
...
+
+>**NOTE:** You need a token with the `broker-upgrade:write` authorization scope to create an orchestration, and a token with the `broker-upgrade:read` scope to fetch the orchestrations.
+
+>**NOTE:** For now, you can upgrade only one Runtime in a single request in KEB.
```
Comment (klaudiagrz) [inline docs/kyma-environment-broker/03-10-orchestration.md:10]:
```suggestion
>**NOTE:** For now, you can upgrade only one Runtime in a single request.
```

File: docs/kyma-environment-broker/03-10-orchestration.md [hunk 19]
```diff
@@ -0,0 +1,18 @@
...
+
+Orchestration API consist of the following handlers:
+
+- **GET** `/orchestrations` - exposes data about a single orchestration status.
+- **GET** `/orchestrations/{orchestration_id}` - exposes data about all the orchestrations.
+- **POST** `/upgrade/kyma` - schedules the orchestration of the Kyma upgrade. Requires specifying a request body.
+
+For more details about API check the [Swagger schema](https://app.swaggerhub.com/apis/kempski/kyma-orchestration_api/0.1).
```
Comment (klaudiagrz) [inline docs/kyma-environment-broker/03-10-orchestration.md:18]:
```suggestion
For more details about the API, check the [Swagger schema](https://app.swaggerhub.com/apis/kempski/kyma-orchestration_api/0.1).
```

File: docs/kyma-environment-broker/03-10-orchestration.md [hunk 20]
```diff
@@ -0,0 +1,18 @@
...
+Orchestration is a mechanism that allows you to upgrade Kyma Runtimes. To create an orchestration, [send the upgrade request to the proper KEB handler](#tutorials-upgrade-kyma-runtime-using-keb). After sending the request, the orchestration is processed by `KymaUpgradeManager`. It lists Shoots (Kyma Runtimes) in the Gardener cluster and filters the Runtime IDs that you have specified for the upgrade in the request body. Then, the `KymaUpgradeManager` proceeds with the orchestration and performs the [upgrade steps](#details-runtime-operations) logic on the Runtimes that you have specified.
+
+>**NOTE:** You need a token with the `broker-upgrade:write` authorization scope to create an orchestration, and a token with the `broker-upgrade:read` scope to fetch the orchestrations.
+
+>**NOTE:** For now, you can upgrade only one Runtime in a single request in KEB.
+
+Orchestration API consist of the following handlers:
+
+- **GET** `/orchestrations` - exposes data about a single orchestration status.
+- **GET** `/orchestrations/{orchestration_id}` - exposes data about all the orchestrations.
```
Comment (klaudiagrz) [inline docs/kyma-environment-broker/03-10-orchestration.md:15]:
```suggestion
- **GET** `/orchestrations/{orchestration_id}` - exposes data about all orchestrations.
```

File: docs/kyma-environment-broker/03-10-orchestration.md [hunk 21]
```diff
@@ -0,0 +1,18 @@
...
+- **POST** `/upgrade/kyma` - schedules the orchestration of the Kyma upgrade. Requires specifying a request body.
```
Comment (klaudiagrz) [inline docs/kyma-environment-broker/03-10-orchestration.md:16]:
```suggestion
- **POST** `/upgrade/kyma` - schedules the orchestration. It requires specifying a request body.
```

Comment (klaudiagrz) [reply docs/kyma-environment-broker/03-10-orchestration.md:16]: I would also unify the formatting of these endpoints 🤔

File: docs/kyma-environment-broker/08-06-orchestration-status.md [hunk 22]
```diff
@@ -0,0 +1,109 @@
...
+          "state": "InProgress" / "Pending" / "Succeeded" / "Failed"
+        }
+        [...]
+      ]
+    }
+   ```
+
+## Fetch all orchestrations
+
+1. Make a call to the Kyma Environment Broker with a proper **Authorization** [request header](#details-authorization) to verify that the orchestration succeeded.
```
Comment (klaudiagrz) [inline docs/kyma-environment-broker/08-06-orchestration-status.md:66]:
```suggestion
Make a call to the Kyma Environment Broker with a proper **Authorization** [request header](#details-authorization) to verify that the orchestration succeeded.
```

**Review by klaudiagrz — APPROVED** [review]

**Pull request merged**